packages/grid/_modules_/grid/GridComponent.tsx

```diff
@@ -81,7 +81,7 @@ export const GridComponent = React.forwardRef<HTMLDivElement, GridComponentProps
     useSorting(apiRef, props.rows);
     useColumnMenu(apiRef);
     usePreferencesPanel(apiRef);
-    useFilter(apiRef);
+    useFilter(apiRef, props.rows);
```

Comment:
Wouldn't it be better if you select the rows from the state? I'm not sure if it will work but that way the only argument needed for the hook is the `apiRef` (we can probably do the same for the others as well if it makes sense).